### PR TITLE
Fix podman-remote info to show registry data

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/libpod/issues/4793

**Comment by rhatdan** [comment]
@marusak  PTAL
@baude @jwhonce @mheon @vrothberg @TomSweeneyRedHat PTAL

File: API.md [hunk 1]
```diff
@@ -1857,6 +1859,14 @@ graph_root [string](https://godoc.org/builtin#string)
 graph_status [InfoGraphStatus](#InfoGraphStatus)
 
 run_root [string](https://godoc.org/builtin#string)
+### <a name="InfoRegistry"></a>type InfoRegistry
+
+InfoRegistry describes the host's registry informatoin
```
Comment (NevilleC) [inline API.md]: s/informatoin/information

**Comment by baude** [comment]
code lgtm

File: pkg/varlinkapi/system.go [hunk 2]
```diff
@@ -90,22 +88,25 @@ func (i *LibpodAPI) GetInfo(call iopodman.VarlinkCall) error {
...
-		insecureRegistriesInterface := info[3].Data["registries"]
-		if insecureRegistriesInterface != nil {
-			insecureRegistries = insecureRegistriesInterface.([]string)
+	if len(info) > 1 {
+		for key, val := range info[2].Data {
+			if key == "search" {
+				podmanInfo.Registries.Search = val.([]string)
+				continue
+			}
+			fmt.Println("DAN4", key, val)
```
Comment (vrothberg) [inline pkg/varlinkapi/system.go]: I think that's a debugging/dev artifact and should be removed.

File: cmd/podman/varlink/io.podman.varlink [hunk 3]
```diff
@@ -245,6 +245,13 @@ type InfoGraphStatus (
     supports_d_type: string
 )
 
+# InfoRegistry describes the host's registry informatoin
```
Comment (vrothberg) [inline cmd/podman/varlink/io.podman.varlink]: s/informatoin/information/

File: pkg/varlinkapi/system.go [hunk 4]
```diff
@@ -90,22 +88,25 @@ func (i *LibpodAPI) GetInfo(call iopodman.VarlinkCall) error {
...
-		if registriesInterface != nil {
-			registries = registriesInterface.([]string)
-		}
-	}
-	if len(info) > 3 {
-		insecureRegistriesInterface := info[3].Data["registries"]
-		if insecureRegistriesInterface != nil {
-			insecureRegistries = insecureRegistriesInterface.([]string)
+	if len(info) > 1 {
+		for key, val := range info[2].Data {
```
Comment (vrothberg) [inline pkg/varlinkapi/system.go]: This  looks suspicious. We check for `> 1` but access the third item which can lead to invalid memory accesses if `len(info) == 2`.

Can we add comments to explain the assumptions of the length check and accesses?

File: pkg/varlinkapi/system.go [hunk 5]
```diff
@@ -90,22 +88,25 @@ func (i *LibpodAPI) GetInfo(call iopodman.VarlinkCall) error {
...
+			if key == "search" {
+				podmanInfo.Registries.Search = val.([]string)
+				continue
+			}
```
Comment (vrothberg) [inline pkg/varlinkapi/system.go]: Why is a search registry treated differently? A search registry could also be marked insecure.

Comment (rhatdan) [reply pkg/varlinkapi/system.go]: libpod/runtime.go sends back two records for registires.

Data type and Search type.  The search type seems to be the old way of sending/storing information in the registries.conf.  

I got this by experimenting around and attempting to make `podman-remote info` look close to `podman info`

File: pkg/varlinkapi/system.go (hunk 4 above)
Comment (rhatdan) [reply pkg/varlinkapi/system.go]: Fixed

File: cmd/podman/varlink/io.podman.varlink (hunk 3 above)
Comment (rhatdan) [reply cmd/podman/varlink/io.podman.varlink]: Fixed

File: pkg/varlinkapi/system.go (hunk 2 above)
Comment (rhatdan) [reply pkg/varlinkapi/system.go]: removed

File: test/e2e/info_test.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-// +build !remoteclient
+// +build
```
Comment (mheon) [inline test/e2e/info_test.go]: Can we remove this entirely if it now allows the remote client?

**Comment by mheon** [comment]
Technically speaking this is a breaking API change, but we've never really managed those before for varlink...

**Comment by rhatdan** [comment]
@mheon @baude Can we get this merged now?

**Pull request merged**